caikit_nlp/modules/text_generation/peft_prompt_tuning.py

```diff
@@ -465,20 +486,41 @@ def train(
                 silence_progress_bars=silence_progress_bars,
                 # NOTE: following can override above arguments in order
                 **filtered_training_arguments,
+                **processing_configuration,
             )
 
-            base_model_trainer = base_model.get_trainer(
-                train_dataset=training_dataset, model=peft_model, **training_args
-            )
 
-            training_loss_history = launch_training(
-                peft_model,
-                training_dataset,
-                training_args,
-                checkpoint_dir,
-                trainer=base_model_trainer,
-                tokenizer=base_model.tokenizer,
-            )
+            if torch.cuda.is_available():
+                # NOTE: torch distributed can hang if run on CPUs,
+                # to avoid that, specially for unit tests, we are only
+                # running below when GPUs are available
+                launch_config = get_torch_elastic_launch_config(
+                    get_config().master_addr,
+                    get_config().master_port,
+                )
+
+                training_loss_history = torch.distributed.launcher.api.elastic_launch(
+                    launch_config, launch_training
+                )(
+                    peft_model,
+                    training_dataset,
+                    training_args,
+                    checkpoint_dir,
+                    base_model
+                )
+                # NOTE: We are currently only storing the loss information from
+                # rank 0, i.e main process. training_loss_history is dictionary containing
+                # rank of the process as key
+                training_loss_history = training_loss_history[0]
+            else:
+                # Do not do FSDP things
+                training_loss_history = launch_training(
+                    peft_model,
+                    training_dataset,
+                    training_args,
+                    checkpoint_dir,
+                    base_model,
```

Comment:
the `launch_training` function on this branch seems to accept `trainer` as the argument after `checkpoint_dir` and is not doing `base_model.get_trainer`. So there seems to be some changes missing in `launch_training` function.

Reply:
Sorry, somehow I forgot to push the updates in one of the files when I was rebasing it! Should be fine now